### PR TITLE
List hosts

### PR DESCRIPTION
**Pull request by ser-io**

### CODE REVIEW

File: app/controller.go [hunk 1]
```diff
@@ -157,6 +158,36 @@ func (c *Controller) CreateHost(w http.ResponseWriter, r *http.Request, user Use
...
+		return err
+	}
+	replyJSON(w, res, http.StatusOK)
+	return nil
+}
+
+func BuildListHostRequest(r *http.Request) (*ListHostsRequest, error) {
+	var maxResults int
+	maxResultsRaw := r.URL.Query().Get("maxResults")
+	if maxResultsRaw != "" {
```
Comment (jemoreira) [inline app/controller.go:177]: might be worth it to encapsulate this in a utility function to parse int query parameters

File: app/gcp/instancemanager.go [hunk 2]
```diff
@@ -104,6 +104,39 @@ func (m *InstanceManager) CreateHost(zone string, req *apiv1.CreateHostRequest,
 	return result, nil
 }
 
+const listHostsRequestMaxResultsLimit = 500
+
+func (m *InstanceManager) ListHosts(zone string, user app.UserInfo, req *app.ListHostsRequest) (*apiv1.ListHostsResponse, error) {
+	if req.MaxResults > listHostsRequestMaxResultsLimit {
```
Comment (jemoreira) [inline app/gcp/instancemanager.go:110]: this doesn't have to be an error, you can return at most listHostsRequestMaxResultsLimit elements and a nextPageToken instead.

**Review by jemoreira — CHANGES_REQUESTED** [review]

File: app/controller.go (hunk 1 above)
Comment (ser-io) [reply app/controller.go]: done

File: app/gcp/instancemanager.go (hunk 2 above)
Comment (ser-io) [reply app/gcp/instancemanager.go]: done

**Review by jemoreira — APPROVED** [review]

**Pull request merged**